src/main/resources/org/nest/sympy/misc.py

```diff
@@ -0,0 +1,178 @@
+    '''The aim of this script is to check for any given function of
+    positive `t`, which we will call `shape`, if it satisfies a linear
+    homogeneous ODE. If it does, it will give it's order, initial
+    values, the equation itself and a list of all factors of
+    derivatives (which we will call `derivative_factors`) of the shape
+    function in the ODE, i.e. if shape''= a*shape+b*shape' the list
+    will be: [a,b].
+
+    Example:
+    ========
+
+    Convert the postsynaptic response shape from a function form  
+       shape = t * exp(t * a)
+    to an ODE form
+       shape'' = -a**2 * shape + 2*a * shape'
+    '''
+
+    def __init__(self, name, function):
+
+        self.name = parse_expr(name)
+
+        
+        # convert the shape function from a string to a symbolic expression
+        shape = parse_expr(function)
+
+        # found_ode is true if we find a linear homogeneous ODE that
+        # `shape` satisfies
+        found_ode = False
+        
+        # First we check if `shape` satisfies a linear homogeneous ODE
+        # of order 1. `derivatives` is a list of all derivatives of `shape`
+        # up to the order we are checking (which we just call 'order')
+        derivatives = [shape, diff(shape, t)]
+        
+        # If `diff_rhs_lhs`, here shape'-derivative_factors*shape=0 for
+        # some 'derivative_factors', 'shape' satisfies a first order
+        # linear homogeneous ODE (and does for t=1). Thereafter, `diff_rhs_lhs will` be the
+        # difference of the derivative of shape of order 'order' and
+        # the sum of all lower derivatives times their
+        # 'derivative_factors which is a list of the potential
+        # factors in the ODE from the factor of shape^(0) to
+        # shape^(order-1).
+        # In the case of the ODE of order 1 we have only one `derivative_factor`
+        # but in all other cases we have several. For unified handling we define
+        # `derivative_factors` as a list for all orders (also for order 1)
+        derivative_factors = (1/derivatives[0] * derivatives[1]).subs(t,1),
+
+        diff_rhs_lhs = derivatives[1] - derivative_factors[0] * derivatives[0]
+
+        if simplify(diff_rhs_lhs) == sympify(0):
+            found_ode = True
+        
+        # Initialize the (potential) order of the differential equation.
+        order = 1
+        
+        # while an ODE has not yet been found and we have not yet
+        # reached the maximum order we are checking for, we check if
+        # `shape` satisfies a linear homogeneous ODE of the next higher
+        # order.
+        while not found_ode and order < MAX_ORDER:
+
+            # The potential order must be at least `order+1`
+            order += 1
+            # Add the next higher derivative to the list of
+            # derivatives of `shape`
+            derivatives.append(diff(derivatives[-1], t))
+            
+            # The goal here is to calculate the factors (which we call
```

Comment:
You can display real LaTeX math with the notebook, which would allow you to write the system that supports your text explanation/code... otherwise it's really hard to follow.